### PR TITLE
support dpg for sdk generation pipeline

### PR DESCRIPTION
**Pull request by zzvswxy**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: tools/azure-sdk-tools/packaging_tools/auto_codegen.py [hunk 1]
```diff
@@ -28,7 +28,7 @@ def main(generate_input, generate_output):
             config = generate(CONFIG_FILE, sdk_folder, [], relative_path_readme, spec_folder, force_generation=True,
                               python_tag=python_tag)
         else:
-            config = gen_dpg(input_readme, data.get('autorestConfig', ''))
+            config = gen_dpg(input_readme, data.get('autorestConfig', ''), "../../../../../azure-rest-api-specs/")
```
Comment (msyyc) [inline tools/azure-sdk-tools/packaging_tools/auto_codegen.py:31]: We could use it as global constant

Comment (zzvswxy) [reply tools/azure-sdk-tools/packaging_tools/auto_codegen.py]: done

File: tools/azure-sdk-tools/packaging_tools/generate_utils.py [hunk 2]
```diff
@@ -162,12 +162,13 @@ def gen_package_name(origin_config: Dict[str, Any]) -> str:
     return Path(origin_config["output-folder"]).parts[-1]
 
 
-def gen_basic_config(origin_config: Dict[str, Any]) -> Dict[str, Any]:
+def gen_basic_config(origin_config: Dict[str, Any], spec_folder: str = "../../../../../azure-rest-api-specs/") -> Dict[str, Any]:
```
Comment (msyyc) [inline tools/azure-sdk-tools/packaging_tools/generate_utils.py:165]: put `DEFAULT_SPEC_FOLDER` in util then you could also use it here

Comment (zzvswxy) [reply tools/azure-sdk-tools/packaging_tools/generate_utils.py]: fixed

**Review by msyyc — APPROVED** [review]

**Pull request merged**